tests/engine/processes/test_builder.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for `aiida.engine.processes.builder.ProcessBuilder`."""
+
+from aiida.calculations.arithmetic.add import ArithmeticAddCalculation
+from aiida.engine.processes.builder import ProcessBuilder
+
+
+def test_access_methods():
```

Comment:
I think we should add more explicit tests that after calling `del` the actual property still works properly. For example directly after deleting it, you can still set it and the validation still works. You are implicitly checking parts of this because you use the same builder, but only for `__setattr__`. Could you use a port  with validation, for example `builder.x`?

Reply:
I'm not sure I understood what you meant here. Can you check my new update? If that was not it, I may need a re-explanation he - he ( ^ - ^ `).